osbuild/api.py

```diff
@@ -40,19 +40,16 @@ class BaseAPI(abc.ABC):
     call `_message.`
     """
 
+    # The name of the API endpoint
+    endpoint: str
```

Comment:
Previously, not overriding the `endpoint()` class method in the child class would result in an error on instantiation of an object. With this change it won't happen, unless you try to use the property.

Reply:
_nodes_, I went through the same links... I admit that there is no ideal solution. There seems to be a way to use `typing.ClassVar` combined with subclassing `typing.Protocol` to catch unset class properties by static analyzers. Unfortunately `typing.Protocol` was added only in Python 3.8 and we need to support 3.6 😞